### PR TITLE
Fix Source Return Bug in Scripting

### LINKED ISSUE #52103: Accessing document _source using Painless script is returning incorrect field value
**Elasticsearch version** (`bin/elasticsearch --version`): 7.5.2

**Plugins installed**: []

**JVM version** (`java -version`): java 13.0.2

**OS version** (`uname -a` if on a Unix-like system):Ubuntu 18.04

**Description of the problem including expected versus actual behavior**:
Painless script is returning incorrect value when accessing document _source

**Steps to reproduce**:
1. Create index with single shard
```
PUT /test
{
    "settings" : {
        "number_of_shards" : 1,
    }
}
```
2. Create documents
```
PUT test/_doc/1
{
    "field" : 1
}
```
```
PUT test/_doc/2
{
    "field" : 2
}
```
3. Query
```
GET test/_search
    {
        "script_fields" : {
            "test1" : {
                "script" : "params['_source']"
            }
        }
    }
```
Expected Result
```
[
  {
    "_index": "test",
    "_type": "_doc",
    "_id": "1",
    "_score": 1,
    "fields": {
      "test1": [
        {
          "field": 1
        }
      ]
    }
  },
  {
    "_index": "test",
    "_type": "_doc",
    "_id": "2",
    "_score": 1,
    "fields": {
      "test1": [
        {
          "field": 2
        }
      ]
    }
  }
]
```
Actual Result
```
[
  {
    "_index": "test",
    "_type": "_doc",
    "_id": "1",
    "_score": 1,
    "fields": {
      "test1": [
        {
          "field": 2
        }
      ]
    }
  },
  {
    "_index": "test",
    "_type": "_doc",
    "_id": "2",
    "_score": 1,
    "fields": {
      "test1": [
        {
          "field": 2
        }
      ]
    }
  }
]
```
Both documents return the same value for script field test1

**Provide logs (if relevant)**:

### PR DESCRIPTION
**Pull request by jdconrad**
This change ensures that when a user returns `_source` directly no matter where accessed within scripting, the value is a Map of the converted source as opposed to a SourceLookup.

Fixes #52103

### CODE REVIEW

File: modules/lang-painless/src/test/java/org/elasticsearch/painless/ScriptedMetricAggContextsTests.java [hunk 1]
```diff
@@ -91,6 +91,33 @@ public void testMapBasic() throws IOException {
...
+        when(lookup.getLeafSearchLookup(leafReaderContext)).thenReturn(leafLookup);
+        SourceLookup sourceLookup = mock(SourceLookup.class);
+        when(leafLookup.asMap()).thenReturn(Collections.singletonMap("_source", sourceLookup));
+        when(sourceLookup.loadSourceIfNeeded()).thenReturn(Collections.singletonMap("test", 1));
+        ScriptedMetricAggContexts.MapScript.LeafFactory leafFactory = factory.newFactory(params, state, lookup);
+        ScriptedMetricAggContexts.MapScript script = leafFactory.newInstance(leafReaderContext);
+
+        script.execute();
+
+        assert(state.containsKey("_source"));
```
Comment (rjernst) [inline modules/lang-painless/src/test/java/org/elasticsearch/painless/ScriptedMetricAggContextsTests.java:116]: I think assertThat with hamcrest matchers will give a nicer error message here if this fails.

**Review by rjernst — APPROVED** [review]
Looks good.

File: modules/lang-painless/src/test/java/org/elasticsearch/painless/ScriptedMetricAggContextsTests.java (hunk 1 above)
Comment (jdconrad) [reply modules/lang-painless/src/test/java/org/elasticsearch/painless/ScriptedMetricAggContextsTests.java]: Fixed.

**Comment by jdconrad** [comment]
@rjernst Thanks for the review! Will commit once CI passes.

**Comment by jdconrad** [comment]
@elasticmachine run elasticsearch-ci/2

**Pull request merged**